platform.md

```diff
@@ -735,43 +736,52 @@ Usage:
 
-A command (`<cmd>`), arguments to that command (`<args>`), a working directory (`<working-dir>`), and an execution strategy (`<direct>`) comprise a process definition. Processes MAY be buildpack-defined or user-defined.
+A command (`<cmd>`), arguments to that command (`<args>`), and a working directory (`<working-dir>`) comprise a process definition. Processes MAY be buildpack-defined or user-defined.
 
 The launcher:
 - MUST derive the values of `<cmd>`, `<args>`, `<working-dir>`, and `<direct>` as follows:
 - **If** the final path element in `$0`, matches the type of any buildpack-provided process type
     - `<process-type>` SHALL be the final path element in `$0`
+    - **If** the buildpack API version supports the `direct` flag on process types, **and** the `direct` flag for this process type is set to `false`
+        - `<direct>` is `false`
+    - **Else**
+        - `<direct>` is `true`.
     - The lifecycle:
         - MUST select the process with type equal to `<process-type>` from `<layers>/config/metadata.toml`
-        - MUST set `<working-dir>` to the value defined for the process in `<layers>/config/metadata.toml`, or to `<app>` if not defined
-        - MUST append any user-provided `<args>` to process arguments
+        - MUST set `<cmd>` to the first element of `command` in the selected process type.
+        - MUST set `<args>` to any remaining elements of `command` in the selected process type, followed by:
+            - **If** there are user-provided `<args>`, the user-provided `<args>`
+            - **Else** any `args` defined in the selected process type
+        - MUST set `<working-dir>` to the `working-dir` defined for the selected process type, or to `<app>` if not defined
 - **Else**
-    - **If** `$1` is `--`
-        - `<direct>` SHALL be `true`
-        - `<cmd>` SHALL be `$2`
-        - `<args>` SHALL be `${@3:}`
-        - `<working-dir>` SHALL be `<app>`
-    - **Else**
-        - `<direct>` SHALL be `false`
-        - `<cmd>` SHALL be `$1`
-        - `<args>` SHALL be `${@2:}`
-        - `<working-dir` SHALL be `<app>`
+    - `<direct>` SHALL be `true`
+    - `<cmd>` SHALL be the user-provided `<cmd>`
+    - `<args>` SHALL be the user-provided `<args>`
+    - `<working-dir>` SHALL be `<app>`
+
+- **If** `<direct>` is `true`
+    - MUST invoke the command `<cmd>` with its arguments `<args>`, environment `<env>`, and working directory `<working-dir>`.
+- **Else** `<direct>` is `false`, and
+    - MUST invoke the command `<cmd>` using a shell with its arguments `<args>`, environment `<env>`, and working directory `<working-dir>`.
+
+- MUST replace all occurrences of `$(<env>)` in `<args>`, where `<env>` is the name of a defined environment variable, with the value of the environment variable after it makes buildpack-provided environment modifications, and before it launches the process.
```

Comment:
```suggestion
- MUST replace all occurrences of `$(<env>)` in `<args>`, where `<env>` is the name of a buildpack-defined environment variable, with the buildpack-provided value, before it launches the process.
```

Reply:
I notice we don't make mention of any mechanism for turning interpolation off completely (https://github.com/buildpacks/rfcs/blob/main/text/0093-remove-shell-processes.md#unresolved-questions) which seems okay to me as I haven't heard anyone specifically ask that we add this - but maybe others have more information than me. cc @ekcasey @samj1912